### PR TITLE
Bugfix/iat 3169

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java [hunk 1]
```diff
@@ -39,11 +48,15 @@ public void migrate() {
...
+            if(this.applicationProperties.isValidateMigrationSets()) {
+                validateMigrationSets();
+            }
+
             List<String> itemIds = this.getItemsToMigrate();
 
             List<CompletableFuture<MigrationItemResult>> futuresList = itemIds.stream()
-                .map(asyncService::migrateDataStoreItemsAsync)
-                .collect(toList());
+                    .map(asyncService::migrateDataStoreItemsAsync)
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java]: As mentioned in the other PR, I use the default intellij formatting and try to only format what I touch but this one got me.

If you want a different one please put the codestyle file on the Getting Started page:
https://www.jetbrains.com/help/idea/copying-code-style-settings.html

File: src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java [hunk 2]
```diff
@@ -39,11 +48,15 @@ public void migrate() {
         if (dataStoreMigrations.isEnabled()) {
             log.info("data store migrations are enabled");
 
+            if(this.applicationProperties.isValidateMigrationSets()) {
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java]: One can now skip this validation check.  I think the check is worth it even if the time it takes is a few minutes considering the damage that would be done if migrations aren't configured correctly.  However, I've now provided a way for deployers to disable it.

File: src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java (hunk 1 above)
Comment (bdrainer) [reply src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java]: I'll put one up when I get a chance.  For the most part, we are going with defaults.  I think the main change is line length.  We use 120 where I think the default is to break at 80.

File: src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java (hunk 2 above)
Comment (bdrainer) [reply src/main/java/org/opentestsystem/ap/migration/service/MigrationService.java]: Agreed.  A few minutes is worth the wait.

File: src/main/java/org/opentestsystem/ap/migration/repository/ItemVersionRepository.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.List;
+
+@Slf4j
+@Repository
+public class ItemVersionRepository {
+    private final String VERSION_QUERY = "SELECT DISTINCT(i.item_json ->> 'version') from ITEM i " +
+            "JOIN (select item_id, max(d.created_date) as \"max_date\" from item d where d.branch_name = 'master' group by d.item_id) as mdj " +
+            "on mdj.item_id = i.item_id and mdj.max_date = i.created_date\n" +
+            "WHERE i.is_deleted = false;";
```
Comment (bdrainer) [inline src/main/java/org/opentestsystem/ap/migration/repository/ItemVersionRepository.java]: I wonder how much faster this is compared to what was there before.  Before it was looking at every record in the database.

Comment (millerts) [reply src/main/java/org/opentestsystem/ap/migration/repository/ItemVersionRepository.java]: Yeah, the issue with the previous query is that you'd get the previous item migration "version".  We only care about the most current "version" value on the latest master record for each item.

**Pull request merged**